windows/src/desktop/kmshell/install/UfrmInstallKeyboardFromWeb.pas

```diff
@@ -215,7 +216,7 @@ procedure TfrmInstallKeyboardFromWeb.DownloadAndInstallPackage(const PackageID,
 begin
   FTempDir := IncludeTrailingPathDelimiter(CreateTempPath);  // I1679
   try
-    FDownloadFilename := FTempDir + PackageID + Ext_PackageFile;
+    FDownloadFilename := FTempDir + TIdURI.URLDecode(PackageID) + Ext_PackageFile;
```

Comment:
Same thing I think, we want PackageID to be a plain string not encoded in any way by the time it gets here, so the cefBeforeBrowseEx function (and any other callers?) would need to decode it.